hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HoodieHiveClient.java

```diff
@@ -319,4 +319,11 @@ public void updateLastCommitTimeSynced(String tableName) {
       throw new HoodieHiveSyncException("Failed to get update last commit time synced to " + lastCommitSynced, e);
     }
   }
+
+  /**
+   * Scan table partitions.
+   */
+  public List<Partition> scanTablePartitions(String tableName) throws TException {
+    return client.listPartitions(syncConfig.databaseName, tableName, (short) -1);
+  }
```

Comment:
the table was passed in as arg but database read from config object. this introduces risk in discrepancy. database and table name should come from the same place: either config object or arguments

Reply:
I think we are passing table name to API calls, since for MoR, we have 2 tables _ro and _rt. But I removed passing the database arg to APIs, and database is always fetched from config, but tables are passed in for private API.

Reply:
ok sg